docs/docs/api-reference/starlark-reference/nix-support.md

```diff
@@ -0,0 +1,53 @@
+---
+title: NixSupport
+sidebar_label: NixSupport
+---
+
+You can provide Kurtosis just with source code and a Nix definition on how to build an image. Kurtosis will take care of building and deploying the image directly into the enclave without the need to upload or register the image beforehand.
+
+For that, we use Nix flakes, which is a way to package build definitions and dependencies in a reproducible manner. Using Nix flakes, you can define your system configurations and dependencies in a single file (`flake.nix`), making it easier to manage and share.
+
+Here's a basic explanation of how you can generate Docker images from services using Nix flakes:
+
+1. **Install Nix**: Installing Nix isn't strictly necessary with Kurtosis, but it's recommended if you are creating or developing the package. You can install it by following the instructions on the Nix website: [https://nixos.org/download.html](https://nixos.org/download.html)
+
+2. **Create a Nix Flake**: Create a directory for your project and initialize a Nix flake. You can do this by running:
```

Comment:
Should we also clarify that this folder should contain the service's source code here? or is it very clear?

Reply:
yeah, not very clear. It is implicitly assuming that you are starting a project and that later you will put the source code there. Maybe a more realistic version would be:
> Go to your project root directory and initialize a Nix flake